app/assets/javascripts/discourse/app/components/user-preferences/categories.hbs

```diff
@@ -9,7 +9,9 @@
         }}</a>
     {{/if}}
     <CategorySelector
+      @categoryIds={{@model.watched_category_ids}}
       @categories={{@model.watchedCategories}}
```

Comment:
Do we still need to pass the categories in when we can derive it from `watched_category_ids`. The only difference is how we derived it will be based on the relevant site settings. Otherwise, I feel like this component is becoming very complicated to use.

Reply:
Unfortunately yes, but only for just a while longer until I can update all usages of `<CategorySelector>` from core, plugins, themes and theme components. After that, I will add a deprecation notice and probably we will be able to remove the old fields in a future release.

Reply:
I think we have to deprecate the `categories` args as part of this change because we can't live in a state where two different yet very similar args exist for a component.

Reply:
I think it is too early to deprecate `categories` because `categoryIds` is currently used just when `lazy_load_categories`, under an experimental site setting.

Reply:
In this case, what do you think about creating a new component that inherits most of the original functionality? In our templates, we will then render the new component conditionally based on the site setting. I think that will be a safer approach than making the existing component more complex.

Reply:
I can make it use either `categoryIds` or `categories` and in the future we can just add a deprecation notice. Can you check the new implementation?

Reply:
Thank you :+1: This looks much better and safer.